pkg/k8s/namespace.go

```diff
@@ -14,4 +15,5 @@ import (
 //	nsl = kubeClient.Namespaces()
 type NamespaceLister interface {
 	List(labels.Selector, fields.Selector) (*api.NamespaceList, error)
+	Watch(label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error)
```

Comment:
@smothiki this should probably go into a separate interface called `NamespaceWatcher` or similar